flipperdevices/flipperzero-firmware
# SubGhz: add protocol BinRAW (binarization of data quantized by the minimum correlated duration)

# What's new

- SubGhz: add protocol BinRAW (binarization of data quantized by the minimum correlated duration)
- SubGhz: add check for free RAM when adding to history
- SubGhz: add display of RSSI level in SubGhz Read
- SubGhz: fix secplus_v2 decoder
- FURI_LOG: add FURI_LOG_RAW_x formatted string output like printf
- SubGhz: fix filter decoder
- SubGhz: change in the LevelDuration structure, the size of the structure occupied in RAM is reduced by 2 times
- SubGhz: forcibly turn off the speaker when exiting SubGhz
- SubGhz: adaptive adjustment to the noise level

# Verification 

- Uncomment the line //#define BIN_RAW_DEBUG
in file lib\subghz\protocols\bin_raw.c
thereby enable the debug output of this protocol. now the protocol will work on all packages received by the flipper.
- Compile and download the firmware, then run SubGhz Read. 
- Accept signals of any protocols that the flipper knows. the decoded signal and the one written to BinRaw will be displayed in the history.
- Save and play the BinRAW signals by taking them on another flipper. The signals must be decoded by the second flipper.
- comment out //#define BIN_RAW_DEBUG, compile, flash, run SubGhz Read. check signal reception, there should be no more BinRAW records
- Go to settings, turn on the DRAW - ON function. now the flipper should receive all the signals and try to decode them. those protocols that he knows will be decoded, those that he does not know will be decoded in BinRAW. (for example, you can play a recording of a weather station or a key fob whose protocol Fliper has not previously decoded). ATTENTION, the correct frequency and modulation settings must be set, if they are not correct, incorrect decoding in BinRAW is possible

# Checklist (For Reviewer)
